go/toolchain/toolchains.bzl

```diff
@@ -1,124 +1,149 @@
-load('//go/private:go_toolchain.bzl', 'external_linker', 'go_toolchain')
+load('//go/private:go_toolchain.bzl', 'go_toolchain')
+load('//go/private:toolchain.bzl', 'go_download_sdk', 'go_host_sdk')
+load('//go/private:go_toolchain.bzl', 'external_linker')
 
 DEFAULT_VERSION = "1.9.1"
 
-def _generate_toolchains():
-  # The set of acceptable hosts for each of the go versions, this is essentially the
-  # set of sdk's we know how to fetch
-  versions = [
-      struct(
-          name = "host",
-          sdk = "@go_host_sdk",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"],
-      ),
-      struct(
-          name = "1.9.1",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"],
-      ),
-      struct(
-          name = "1.9",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"],
-      ),
-      struct(
-          name = "1.8.4",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"],
-      ),
-      struct(
-          name = "1.8.3",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"],
-      ),
-      struct(
-          name = "1.8.2",
-          hosts = ["darwin_amd64", "linux_amd64"],
-      ),
-      struct(
-          name = "1.8.1",
-          hosts = ["darwin_amd64", "linux_amd64"],
-      ),
-      struct(
-          name = "1.8",
-          hosts = ["darwin_amd64", "linux_amd64"],
-      ),
-      struct(
-          name = "1.7.6",
-          hosts = ["darwin_amd64", "linux_386", "linux_amd64", "windows_386", "windows_amd64", "freebsd_386", "freebsd_amd64"]
-      ),
-      struct(
-          name = "1.7.5",
-          hosts = ["darwin_amd64", "linux_amd64"],
-      ),
-  ]
+HOSTS = (
```

Comment:
Consider renaming to `HOST_TARGETS` or something that expresses that this is a map from host platforms to target platforms.